sonarts-core/src/se/stateTransitions.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * SonarTS
+ * Copyright (C) 2017-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as ts from "typescript";
+import * as tsutils from "tsutils";
+import {
+  createLiteralSymbolicValue,
+  createUnknownSymbolicValue,
+  createUndefinedSymbolicValue,
+  createObjectLiteralSymbolicValue,
+} from "./symbolicValues";
+import { ProgramState } from "./programStates";
+
+export function applyExecutors(programPoint: ts.Node, state: ProgramState, program: ts.Program): ProgramState {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') The function applyExecutors has a cyclomatic complexity of 11 which is higher than the threshold of 10 [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS1541)